test/algorithms/test_wstate.cpp

```diff
@@ -0,0 +1,58 @@
+#include "algorithms/WState.hpp"
+#include "dd/FunctionalityConstruction.hpp"
+#include "dd/Simulation.hpp"
+
+#include "gtest/gtest.h"
+#include <iostream>
+#include <vector>
+
+class WState : public testing::TestWithParam<std::size_t> {
+protected:
+  void TearDown() override {}
+  void SetUp() override {}
+};
+
+std::vector<std::string> generateWStateStrings(std::size_t length) {
+  std::vector<std::string> result;
+  for (int i = 0; i < (1 << length); i++) {
+    int countOnes = 0;
+    std::string binaryString;
+    for (size_t j = 0; j < length; j++) {
+      if (((i >> j) & 1) != 0) {
+        countOnes++;
+        binaryString += "1";
+      } else {
+        binaryString += "0";
+      }
+    }
+    if (countOnes == 1) {
+      result.push_back(binaryString);
+    }
+  }
+  return result;
+}
```

Comment:
```suggestion
std::vector<std::string> generateWStateStrings(const std::size_t length) {
  std::vector<std::string> result;
  result.reserve(length);
  for (std::size_t i = 0U; i < length; ++i) {
    auto binaryString = std::string('0', length);
    binaryString[i] = '1';
    result.emplace_back(binaryString);
  }
  return result;
}
```
Way simpler and more scalable. Also makes sure to preallocate the right size for the vector.